docarray/utils/create.py

```diff
@@ -0,0 +1,220 @@
+from docarray import DocList, BaseDoc
+from docarray.typing import AnyTensor
+from pydantic import create_model
+from typing import Dict, List, Any, Union, Optional, Type
+
+
+def create_new_model_cast_doclist_to_list(model: Any) -> BaseDoc:
+    """
+    Take a Pydantic model and cast DocList fields into List fields.
+
+    This can be needed because of this limitation of Pydantic:
+
+    (https://docs.pydantic.dev/latest/blog/pydantic-v2/)
+
+    ---
+
+    ```python
+    from docarray import BaseDoc
+    class MyDoc(BaseDoc):
+        tensor: Optional[AnyTensor]
+        url: ImageUrl
+        title: str
+        texts: DocList[TextDoc]
+
+    MyDocCorrected = create_new_model_cast_doclist_to_list(CustomDoc)
+    ```
+
+    ---
+    :param model: The input model
+    :return: A BaseDoc class dynamically created with List instead of DocList in the schema.
+    """
+    fields: Dict[str, Any] = {}
+    for field_name, field in model.__annotations__.items():
+        try:
+            if issubclass(field, DocList):
+                t: Any = field.doc_type
+                fields[field_name] = (List[t], {})
+            else:
+                fields[field_name] = (field, {})
+        except TypeError:
+            fields[field_name] = (field, {})
+    return create_model(
+        model.__name__, __base__=model, __validators__=model.__validators__, **fields
+    )
+
+
+def _get_field_from_type(
+    field_schema,
+    field_name,
+    root_schema,
+    cached_models,
+    is_tensor=False,
+    num_recursions=0,
+):
```

Comment:
Can we have type hints to that mypy can do its job?

A small docstring just for internal purposes would also be nice, the code is quite complicated.